cmd/kpod/history.go

```diff
@@ -0,0 +1,349 @@
+		outputSize = units.HumanSize(float64(params.Size))
+	}
+
+	// templParams is used to store the info from params and the time and
+	// size that have been converted to type string for when the human flag
+	// is set
+	templParams := struct {
+		ID        string
+		Created   string
+		CreatedBy string
+		Size      string
+		Comment   string
+	}{
+		params.ID,
+		createdTime,
+		params.CreatedBy,
+		outputSize,
+		params.Comment,
+	}
+
+	if err = templ.Execute(os.Stdout, templParams); err != nil {
+		return err
+	}
+	fmt.Println()
+	return nil
+}
+
+// outputTime displays the time stamp in "2017-06-20T20:24:10Z" format
+func outputTime(tm *time.Time) string {
+	return tm.Format(time.RFC3339)
+}
+
+// outputHumanTime displays the time elapsed since creation
+func outputHumanTime(tm *time.Time) string {
+	return units.HumanDuration(time.Since(*tm))
+}
+
+// createJSON retrieves the history of the image and returns a JSON object
+func createJSON(store storage.Store, opts historyOptions) ([]byte, error) {
+	var (
+		size     int64
+		img      *storage.Image
+		imageID  string
+		layerAll []historyOutputParams
+	)
+
+	ref, err := is.Transport.ParseStoreReference(store, opts.image)
+	if err != nil {
+		return nil, errors.Errorf("error parsing reference to image %q: %v", opts.image, err)
+	}
+
+	img, err = is.Transport.GetStoreImage(store, ref)
+	if err != nil {
+		return nil, errors.Errorf("no such image %q: %v", opts.image, err)
+	}
+
+	systemContext := getSystemContext("")
+
+	src, err := ref.NewImage(systemContext)
+	if err != nil {
+		return nil, errors.Errorf("Error instantiating image %q: %v", opts.image, err)
+	}
+
+	oci, err := src.OCIConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	history := oci.History
+	layers := src.LayerInfos()
+	count := 1
+
+	// iterating backwards to get newwest to oldest
+	for i := len(history) - 1; i >= 0; i-- {
+		if i == len(history)-1 {
+			imageID = img.ID
+		} else {
+			imageID = "<missing>"
+		}
+
+		if !history[i].EmptyLayer {
+			size = layers[len(layers)-count].Size
+			count++
+		} else {
+			size = 0
+		}
+
+		params := historyOutputParams{
+			ID:        imageID,
+			Created:   history[i].Created,
+			CreatedBy: history[i].CreatedBy,
+			Size:      size,
+			Comment:   history[i].Comment,
+		}
+
+		layerAll = append(layerAll, params)
+	}
+
+	output, err := json.MarshalIndent(layerAll, "", "\t\t")
+	if err != nil {
+		return nil, errors.Errorf("Error marshalling to JSON: %v", err)
+	}
+
+	if err = src.Close(); err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+// outputHistory gets the history of the image from the JSON object
+// and pretty prints it to the screen
+func outputHistory(store storage.Store, opts historyOptions) error {
+	var (
+		outputCreatedBy string
+		imageID         string
+		history         []historyOutputParams
+	)
+
+	raw, err := createJSON(store, opts)
+	if err != nil {
+		return errors.Errorf("Error creating JSON: %v", err)
```

Comment:
errors that aren't logged should start with a lowercase (all throughout this PR).